### PR TITLE
[feat] Refactor training framework into fastvideo/train

### PR DESCRIPTION
**Pull request by alexzms**
## Summary

  Introduces `fastvideo/train`, a refactored training framework that replaces the monolithic training/distillation pipelines with a modular, YAML-driven architecture.

  ### Key design changes

  - **`_target_`-based instantiation**: Models and methods are selected via `_target_` keys in YAML (e.g., `fastvideo.train.models.wan.WanModel`,
  `fastvideo.train.methods.distribution_matching.dmd2.DMD2Method`), making it easy to add new models/methods without modifying framework code.
  - **Separated concerns**: Models (`models/`), methods (`methods/`), callbacks (`callbacks/`), and the training loop (`trainer.py`) are fully decoupled. The trainer calls
  `method.train_one_step()` without knowing which method is running.
  - **Callback system**: Gradient clipping, validation, and EMA are now callbacks (`callbacks/`) rather than hardcoded in the training loop. Configured via the `callbacks:`
  section in YAML.
  - **Structured config with defaults**: `TrainingConfig` dataclass (`utils/training_config.py`) provides typed defaults for all training parameters. The fully-resolved config
  (with defaults filled in) is logged to W&B.
  - **Checkpoint management**: DCP-based save/resume with `CheckpointManager`, plus `dcp_to_diffusers.py` for converting checkpoints to Diffusers format.

  ### Supported models & methods

  | Models | Methods |
  |--------|---------|
  | Wan 2.1 (T2V 1.3B) | DMD2 distillation |
  | WanGame (incl. causal) | Self-forcing distillation |
  | | SFT finetuning |
  | | DFSFT (Diffusion ForcingSFT) |

  ### Bug fixes

  - **CFG formula**: Fixed `real_score_guidance_scale` in DMD2 and self-forcing to use the standard formula `uncond + scale * (cond - uncond)` instead of `cond + scale * (cond -
  uncond)` (which silently added +1 to the effective guidance scale).

  ### File structure

  fastvideo/train/
    trainer.py
    models/{base, wan/, wangame/}
    methods/{base, distribution_matching/, fine_tuning/}
    callbacks/{callback, grad_clip, validation, ema}
    entrypoint/{train, dcp_to_diffusers}
    utils/{config, builder, training_config, checkpoint, dataloader, optimizer, tracking, ...}

  ### Usage

  ```bash
  torchrun --nproc_per_node=8 -m fastvideo.train.entrypoint.train \
      --config examples/distillation/refactor/distill_wan2.1_t2v_1.3B_dmd2.yaml

  Test plan

  - DMD2 8-step distillation on Wan 2.1 T2V 1.3B matches legacy training loss curves
  - VSA finetuning on Wan produces equivalent results to legacy pipeline
  - Self-forcing distillation on WanGame runs without errors
  - DFSFT on WanGame runs without errors
  - Checkpoint save/resume round-trips correctly
  - W&B logging shows fully-resolved config with defaults

### CODE REVIEW

**Comment by alexzms** [comment]
Related Issue: https://github.com/hao-ai-lab/FastVideo/issues/1158
